### PR TITLE
[cli] "ipmaddr" add with multiple addresses

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
This commit extends "ipmaddr add" command with possibility to use multiple addresses.

This is proposal change to address [DEV-2163 ](https://threadgroup.atlassian.net/browse/DEV-2163?focusedCommentId=50876).

### CODE REVIEW

File: include/openthread/ip6.h [hunk 1]
```diff
@@ -335,7 +335,7 @@ const otNetifAddress *otIp6GetUnicastAddresses(otInstance *aInstance);
  *                                 addresses.
  *
  */
-otError otIp6SubscribeMulticastAddress(otInstance *aInstance, const otIp6Address *aAddress);
+otError otIp6SubscribeMulticastAddress(otInstance *aInstance, const otIp6Address *aAddress, uint8_t aAddressNum);
```
Comment (simonlingoogle) [inline include/openthread/ip6.h:338]: I don't think it has any difference to register multiple addresses at the same time. 

I mean the underlying method `Netif::SubscribeExternalMulticast` still has to register addresses one by one. 
So the two ways make no actual difference:
- Register multiple MAs with one `otIp6SubscribeMulticastAddress` call
- Register multiple MAs with multiple `otIp6SubscribeMulticastAddress` calls, one call for each MA

So, I think we don't actually need to change the API and core implementation, even if we want to support `ipmaddr add` multiple MAs. 

Thoughts?

Comment (lmaciejonczyk) [reply include/openthread/ip6.h:338]: I thought with my changes `Netif::SubscribeExternalMulticast` registers multicast addresses with one /n/mr package. Doesn't it create /n/mr message based on `mMulticastAddresses` list of addresses to register? If not how to achieve registering multiple addresses in one package?

Comment (simonlingoogle) [reply include/openthread/ip6.h:338]: > I thought with my changes Netif::SubscribeExternalMulticast registers multicast addresses with one /n/mr package.

I think as long as we subscribe multiple MAs in one `ipmaddr add` command, there should be only one /n/mr message. The `otIp6SubscribeMulticastAddress` API can be kept as is (i.e. subscribe to one MA at a time), but we can call `otIp6SubscribeMulticastAddress` multiple times in one `ipmaddr add` command processing. 

Going into further details, multiple calls to `otIp6SubscribeMulticastAddress` will  trigger multiple `Get<Notifier>().Signal(kEventIp6MulticastSubscribed)`, which will be merged to one `kEventIp6MulticastSubscribed` event, which eventually triggered the `/n/mr` message containing all the MAs.

**Comment by EskoDijk** [comment]
I've just created an alternative PR for this: https://github.com/openthread/openthread/pull/7253 , without requiring API changes.

**Comment by simonlingoogle** [comment]
Should close it in favor of https://github.com/openthread/openthread/pull/7253?

**Comment by lmaciejonczyk** [comment]
I'm closing this PR in favor of #7253

**Pull request closed without merging**